packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -336,6 +372,7 @@ export const queryBarReducer: Reducer<QueryBarState> = (
   if (isAction<ApplyQueryAction>(action, QueryBarActions.ApplyQuery)) {
     return {
       ...state,
+      fields: action.fields,
```

Comment:
Reducers should be deriving state, not actions: https://redux.js.org/style-guide#put-as-much-logic-as-possible-in-reducers